File: src/instruction.rs
use borsh::{BorshSerialize, BorshDeserialize};
use solana_program::borsh::try_from_slice_unchecked;

use crate::state::Class;

#[derive(BorshSerialize, BorshDeserialize)]
pub enum InstructionEnum {
    MintNft(Class),
    MintNewCollection,
    Redeem,
    ImprintRarity,
    AllocateSol,
    DeAllocateSol,
    DelegateSol,
    UnDelegateSol,
}


impl InstructionEnum {
    pub fn decode(data: &[u8]) -> Self {
        try_from_slice_unchecked(data).unwrap()
    }
}

File: python/instruction.py
from borsh_construct import *
from state import *

InstructionEnum = Enum(
    "MintNft",
    "MintNewCollection",
    "Redeem",
    "ImprintRarity",
    "AllocateSol",
    "DeAllocateSol",
    "DelegateSol",
    "UndelegateSol",
    
    enum_name = "InstructionEnum",
)

def build_instruction(instruction, value = None):
    if instruction == "MintNft":
        return InstructionEnum.build(InstructionEnum.enum.MintNft()) +  ClassEnum.build(value)
    elif instruction == "MintNewCollection":
        return InstructionEnum.build(InstructionEnum.enum.MintNewCollection())
    elif instruction == "Redeem":
        return InstructionEnum.build(InstructionEnum.enum.Redeem())
    elif instruction == "ImprintRarity":
        return InstructionEnum.build(InstructionEnum.enum.ImprintRarity())
    elif instruction == "AllocateSol":
        return InstructionEnum.build(InstructionEnum.enum.AllocateSol())
    elif instruction == "DeAllocateSol":
        return InstructionEnum.build(InstructionEnum.enum.DeAllocateSol())
    elif instruction == "DelegateSol":
        return InstructionEnum.build(InstructionEnum.enum.DelegateSol())
    elif instruction == "UndelegateSol":
        return InstructionEnum.build(InstructionEnum.enum.UndelegateSol())